### PR TITLE
typing: improved mypy support

### LINKED ISSUE #305: Typing support for the public API
Hi! Thanks a lot for building this awesome tool!

I am wondering if providing `typing` support is on your list?
It would be great to have `mypy` checked code, especially in async code.

This is one of the main reasons I look for better celery alternatives. They don't have types and there's no plan to support them. It would be awesome to have at least user-facing API typed and [pep-0561](https://www.python.org/dev/peps/pep-0561/) compatible.

I can probably help with adding some basic stuff 👍

### PR DESCRIPTION
**Pull request by staticdev**
* py.typed mark as required by [PEP-561](https://peps.python.org/pep-0561/)
* add mypy.ini config
* fix JSONEncoder error handling
* some mypy error fixes
* some isort error fixes

Closes #305

### CODE REVIEW

**Comment by staticdev** [comment]
@Bogdanp there is still a lot of typing missing, but this PR will remove some complaints from mypy of when you import dramatiq.

File: dramatiq/broker.py [hunk 1]
```diff
@@ -23,7 +25,7 @@
 global_broker = None
 
 
-def get_broker() -> "Broker":
```
Comment (CaselIT) [inline dramatiq/broker.py:26]: I believe this cannot return `None`

File: dramatiq/results/backends/stub.py [hunk 2]
```diff
@@ -29,7 +30,7 @@ class StubBackend(ResultBackend):
         result data.  Defaults to :class:`.JSONEncoder`.
     """
 
-    results = {}
+    results: Dict[str, Tuple[Optional[str], Optional[str]]] = {}
```
Comment (CaselIT) [inline dramatiq/results/backends/stub.py:33]: the second value of the tuple is a float

**Review by CaselIT — CHANGES_REQUESTED** [review]

File: dramatiq/results/backends/stub.py (hunk 2 above)
Comment (staticdev) [reply dramatiq/results/backends/stub.py]: Thanks for spotting that. Fixed!

File: dramatiq/broker.py (hunk 1 above)
Comment (staticdev) [reply dramatiq/broker.py:26]: @CaselIT I think I found a better way out.. if we cast the type of global_broker to "Broker" it now stop type-checking errors and enables proper evaluation of the global variable.

Comment (staticdev) [reply dramatiq/broker.py]: Fixed also.

**Comment by staticdev** [comment]
LGTM? @CaselIT @Bogdanp

**Review by CaselIT — APPROVED** [review]

**Comment by staticdev** [comment]
@Bogdanp could we get a new release with latests commits?

**Pull request merged**